p2p/agent.go

```diff
@@ -542,15 +580,16 @@ func (p *agent) connect(ctx context.Context) error {
 }
 
 func (p *agent) reconnect() {
-	peers, err := p.Neighbors(context.Background())
-	if err == ErrAgentNotStarted {
+	if p.host == nil {
 		return
 	}
-	if len(peers) == 0 || p.qosMetrics.lostConnection() {
+	if len(p.host.ConnectedPeers()) == 0 || p.qosMetrics.lostConnection() {
 		log.L().Info("network lost, try re-connecting.")
 		p.host.ClearBlocklist()
-		p.connect(context.Background())
+		p.joinP2P(context.Background())
+		return
 	}
+	p.host.FindPeers(context.Background())
```

Comment:
there is also a `context.Background()` at line 589, use one ctx

Reply:
this is not an issue. the func is directly returned in L590

Reply:
yes, the code can work fine. But programming-wise, feels better to define a common `ctx = context.Background()` ahead, and use it at 2 places, rather than declare a new context wherever needed.